modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,68 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  it('TINY-8641: Dispatch change and set editor dirty if the current content and the undoManager history are incoherent', () => {
+    const editor = hook.editor();
+
+    let changeEventCounter = 0;
+    let currentChangeEvent: { level: string; lastLevel: string };
+
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevel = 'a modified last level';
+
+    const assertChangeEvent = (event: { level: string; lastLevel: string }) => {
+      assert.equal(event.level, initialContent, 'Level has not the expected content');
+      assert.equal(event.lastLevel, manualModifiedLevel, 'Last level has not the expected content');
+    };
+
+    editor.resetContent(initialContent);
+    const onChange = (e: EditorEvent<{
+      level: UndoLevel;
+      lastLevel: UndoLevel;
+    }>) => {
+      changeEventCounter++;
+      currentChangeEvent = { level: e.level.content, lastLevel: e.lastLevel.content };
+    };
+
+    editor.on('change', onChange);
+
+    assert.equal(changeEventCounter, 0, '0 event should be detected at start');
+    editor.undoManager.dispatchIfChanged();
+    assert.equal(changeEventCounter, 0, '0 event should be detected if content is coherent with the undoManager history');
+
+    Arr.last(editor.undoManager.data).each((lastLevel) => {
+      lastLevel.content = manualModifiedLevel;
+    });
+
+    assert.equal(editor.isDirty(), false, 'Editor should not be dirty before dispatchIfChanged');
+    editor.undoManager.dispatchIfChanged();
+    assertChangeEvent(currentChangeEvent);
+    assert.equal(changeEventCounter, 1, '1 event should be detected if the content is not coherent with the undoManager history');
```

Comment:
```suggestion
    assert.equal(changeEventCounter, 1, '1 event should be detected if the content is not consistent with the undoManager history');
```

Reply:
done, I also replaced the terms `coherent` and `incoherent`, in those 2 places: [1](https://github.com/tinymce/tinymce/pull/7840/files#diff-a02245522e6da74182ef7dede5d329ddcce06854c05146e4d4abc9194c4d4ea8R645), [2](https://github.com/tinymce/tinymce/pull/7840/files#diff-a02245522e6da74182ef7dede5d329ddcce06854c05146e4d4abc9194c4d4ea8R654)